integration/auth/user_mgt_test.go

```diff
@@ -91,6 +92,258 @@ func TestGetNonExistingUser(t *testing.T) {
+		if err != nil {
+			t.Fatalf("GetUsers([valid identifiers]) returned an error: %v", err)
+		}
+
+		if !sameUsers(getUsersResult.Users, []string{"uid1", "uid2", "uid3", "uid4"}) {
+			t.Errorf("GetUsers([valid identifiers]) = %v; want = %v (in any order)",
+				getUsersResult.Users, []string{"uid1", "uid2", "uid3", "uid4"})
+		}
+	})
+
+	t.Run("returns found users and ignores non-existing users", func(t *testing.T) {
+		getUsersResult, err := client.GetUsers(context.Background(), []auth.UserIdentifier{
+			auth.UIDIdentifier{UID: "uid1"},
+			auth.UIDIdentifier{UID: "uid_that_doesnt_exist"},
+			auth.UIDIdentifier{UID: "uid3"},
+		})
+		if err != nil {
+			t.Fatalf("GetUsers([...]) returned an error: %v", err)
+		}
+
+		if !sameUsers(getUsersResult.Users, []string{"uid1", "uid3"}) {
+			t.Errorf("GetUsers([valid identifiers]) = %v; want = %v (in any order)",
+				getUsersResult.Users, []string{"uid1", "uid3"})
+		}
+		if len(getUsersResult.NotFound) != 1 {
+			t.Errorf("len(GetUsers([...]).NotFound) = %d; want 1", len(getUsersResult.NotFound))
+		} else {
+			if getUsersResult.NotFound[0].(auth.UIDIdentifier).UID != "uid_that_doesnt_exist" {
+				t.Errorf("GetUsers([...]).NotFound[0].UID = %s; want 'uid_that_doesnt_exist'",
+					getUsersResult.NotFound[0].(auth.UIDIdentifier).UID)
+			}
+		}
+	})
+
+	t.Run("returns nothing when queried for only non-existing users", func(t *testing.T) {
+		getUsersResult, err := client.GetUsers(context.Background(), []auth.UserIdentifier{
+			auth.UIDIdentifier{UID: "non-existing user"},
+		})
+		if err != nil {
+			t.Fatalf("GetUsers([valid identifiers]) returned an error: %v", err)
+		}
+
+		if len(getUsersResult.Users) != 0 {
+			t.Errorf("len(GetUsers([...]).Users) = %d; want = 0", len(getUsersResult.Users))
+		}
+		if len(getUsersResult.NotFound) != 1 {
+			t.Errorf("len(GetUsers([...]).NotFound) = %d; want = 1", len(getUsersResult.NotFound))
+		} else {
+			if getUsersResult.NotFound[0].(auth.UIDIdentifier).UID != "non-existing user" {
+				t.Errorf("GetUsers([...]).NotFound[0].UID = %s; want 'non-existing user'",
+					getUsersResult.NotFound[0].(auth.UIDIdentifier).UID)
+			}
+		}
+	})
+
+	t.Run("de-dups duplicate users", func(t *testing.T) {
+		getUsersResult, err := client.GetUsers(context.Background(), []auth.UserIdentifier{
+			auth.UIDIdentifier{UID: "uid1"},
+			auth.UIDIdentifier{UID: "uid1"},
+		})
+		if err != nil {
+			t.Fatalf("GetUsers([valid identifiers]) returned an error: %v", err)
+		}
+
+		if len(getUsersResult.Users) != 1 {
+			t.Errorf("len(GetUsers([...]).Users) = %d; want = 1", len(getUsersResult.Users))
+		} else {
+			if getUsersResult.Users[0].UID != "uid1" {
+				t.Errorf("GetUsers([...]).Users[0].UID = %s; want = 'uid1'", getUsersResult.Users[0].UID)
+			}
+		}
+		if len(getUsersResult.NotFound) != 0 {
+			t.Errorf("len(GetUsers([...]).NotFound) = %d; want = 0", len(getUsersResult.NotFound))
+		}
+	})
+
+	t.Run("returns users with a LastRefreshTimestamp", func(t *testing.T) {
```

Comment:
This is not really related to GetUsers() if I understood correct. May be define as a separate top-level test case. Or see if we can add the relevant checks to the existing `TestCreateUser`  test case.

Reply:
Shouldn't this be removed?

Reply:
Yup. :)